### PR TITLE
Added checks for End of Time during Non-Term dates.

### PR DESCRIPTION
**Pull request by mstratford**
Stops 2016-site from breaking when not on air.

### CODE REVIEW

File: timeslot.go [hunk 1]
```diff
@@ -64,14 +64,31 @@ func (s *Session) GetCurrentAndNext() (*CurrentAndNext, error) {
...
+		nextEndTime, err = strconv.ParseInt(currentAndNext.Next.EndTimeRaw, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		currentAndNext.Next.EndTime = time.Unix(currentEndTime, 0)
+	}
+
 	currentAndNext.Current.StartTime = time.Unix(currentAndNext.Current.StartTimeRaw, 0)
-	currentAndNext.Current.EndTime = time.Unix(currentAndNext.Current.EndTimeRaw, 0)
+	//currentAndNext.Current.EndTime = time.Unix(currentAndNext.Current.EndTimeRaw, 0)
```
Comment (ChrisTheBaron) [inline timeslot.go]: I would remove these lines, not just comment them out 😛

Comment (mstratford) [inline timeslot.go]: Done :)

**Comment by mstratford** [comment]
Bump so that I can enjoy the "Not Scheduled" fix on my install too? 👍

File: timeslot.go [hunk 2]
```diff
@@ -64,14 +64,29 @@ func (s *Session) GetCurrentAndNext() (*CurrentAndNext, error) {
 		return nil, err
 	}
 	var currentAndNext CurrentAndNext
+	var currentEndTime int64
+	var nextEndTime int64
```
Comment (LordAro) [inline timeslot.go]: Why are these declared outside the if statement? Declare on first use

File: timeslot.go [hunk 3]
```diff
@@ -64,14 +64,29 @@ func (s *Session) GetCurrentAndNext() (*CurrentAndNext, error) {
...
 		return nil, err
 	}
+	if currentAndNext.Current.EndTimeRaw != "The End of Time" && (currentAndNext.Current.EndTimeRaw != "") {
+		currentEndTime, err = strconv.ParseInt(currentAndNext.Current.EndTimeRaw, 10, 64)
+		if err != nil {
+			return nil, err
+		}
+		currentAndNext.Current.EndTime = time.Unix(currentEndTime, 0)
+	}
+	if (currentAndNext.Next.EndTimeRaw != "The End of Time") && (currentAndNext.Next.EndTimeRaw != "") {
```
Comment (LordAro) [inline timeslot.go]: These 2 blocks seem oddly similar. Thought about abstracting them into a function?

**Comment by mstratford** [comment]
Note to self: "json: cannot unmarshal number into Go value of type string" loading home page.

**Pull request closed without merging**